processor/tailsamplingprocessor/internal/cache/nop_cache.go

```diff
@@ -0,0 +1,24 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cache // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor/internal/cache"
+
+import "go.opentelemetry.io/collector/pdata/pcommon"
+
+type nopDecisionCache[V any] struct{}
+
+var _ Cache[any] = (*nopDecisionCache[any])(nil)
+
+func NewNopDecisionCache[V any]() Cache[V] {
+	return &nopDecisionCache[V]{}
+}
+
+func (n *nopDecisionCache[V]) Get(_ pcommon.TraceID) (V, bool) {
+	var v V
+	return v, false
```

Comment:
that's interesting -- you are returning the zero value of V, instead of nil, right? wouldn't it be more appropriate to return nil, as in, "not found"?

Reply:
Because it's generic, you can't return nil. Because, for example the type was a primitive or not a reference type, it can't be nil.